server/sql.go

```diff
@@ -159,9 +169,50 @@ func (s *Server) registerSQLRoutes(g *echo.Group) {
 		adviceList := []advisor.Advice{}
 
 		if s.feature(api.FeatureSchemaReviewPolicy) && api.IsSchemaReviewSupported(instance.Engine) {
-			adviceLevel, adviceList, err = s.sqlCheck(ctx, instance, exec)
+			dbType, err := api.ConvertToAdvisorDBType(instance.Engine)
 			if err != nil {
-				return err
+				return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to convert db type %v into advisor db type", instance.Engine))
+			}
+
+			databaseFind := &api.DatabaseFind{
+				InstanceID: &instance.ID,
+				Name:       &exec.DatabaseName,
+			}
+			dbList, err := s.store.FindDatabase(ctx, databaseFind)
+			if err != nil {
+				return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch database `%s` for instance ID: %d", exec.DatabaseName, instance.ID)).SetInternal(err)
+			}
+			if len(dbList) == 0 {
+				return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Database `%s` for instance ID: %d not found", exec.DatabaseName, instance.ID))
+			}
+			if len(dbList) > 1 {
+				return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("There are multiple database `%s` for instance ID: %d", exec.DatabaseName, instance.ID))
+			}
+			db := dbList[0]
+
+			adviceLevel, adviceList, err = s.sqlCheck(
+				ctx,
+				dbType,
+				db.CharacterSet,
+				db.Collation,
+				instance.EnvironmentID,
+				exec.Statement,
+				store.NewCatalog(&db.ID, s.store),
+			)
+			if err != nil {
+				if e, ok := err.(*common.Error); ok && e.Code == common.NotFound {
```

Comment:
I feel this should be handled inside s.sqlCheck

Reply:
I'm considering the `s.sqlCheck` as the util to exec the SQL check. The returned error will be treated differently in different controllers.
In `POST /sql/execute`, if the error is `NotFound`, we just skip the check and ignore the error;
but in `GET /sql/advise`, seems we should return `BadRequest` if cannot find the schema review policy?

Reply:
I am actually thinking in both cases, we return OK and suggest in the return result that no policy configured.

Reply:
No problem. Updated